Review and update Install doc

lread opened the pull request:
Formatted overview on ways to install to a table.
For me this works better to compare variations.
It also helped me to fill in some missing blanks.
And to remove some details unnecessary for an overview (I moved some of these details down to specific install sections).
Let me know if this works for you too.

Noticed that overview mentions how to switch versions but install instructions do not talk about this. Did nothing about this yet.

Subtle: tried to use "we" only when talking from Polylith core team, and used "you" when talking to and guiding the reader.

Assumed poly core team preference is poly as stand-alone and/or Clojure CLI dep. Can adjust advice if this is not the case.
But making some clear suggestion avoids overwhelming newcomers.

Noticed some prerequisite and post-install advice repeated in some install instructions. But inconsistently. I moved these to adoc custom attributes so we have less copy/paste.

Now using adoc `subs="attributes+"` option in code blocks. This way code blocks can now reflect the `:poly-version:` attribute.

Moved clojure cli dep install instructions before clojure tools install instructions. This matches the overview order, and it (maybe?) the more likely order of interest for readers.

Added "stand-alone" to all headings that describe stand-alone installation. Just to make it extra clear.

Moved logging deps config to tools deps usage section as it only applies to that usage.

Re-orged Clojure Tools instructions a bit.
Clarified, I think, how it processes args.

Noticed that JVM options only applies to stand-alone variants to added that detail. I also noticed that it does not give an example, which I thought would be useful, but did nothing about that.

lread committed:
Review and update Install doc

Formatted overview on ways to install to a table.
For me this works better to compare variations.
It also helped me to fill in some missing blanks.
And to remove some details unnecessary for an overview (I moved some of
these details down to specific install sections).
Let me know if this works for you too.

Noticed that overview mentions how to switch versions but install
instructions do not talk about this. Did nothing about this yet.

Subtle: tried to use "we" only when talking from Polylith core team, and
used "you" when talking to and guiding the reader.

Assumed poly core team preference is poly as stand-alone and/or Clojure CLI dep.
Can adjust advice if this is not the case.
But making some clear suggestion avoids overwhelming newcomers.

Noticed some prerequisite and post-install advice repeated in some
install instructions. But inconsistently. I moved these to adoc custom
attributes so we have less copy/paste.

Now using adoc `subs="attributes+"` option in code blocks.
This way code blocks can now reflect the `:poly-version:` attribute.

Moved clojure cli dep install instructions before clojure tools
install instructions. This matches the overview order, and it (maybe?)
the more likely order of interest for readers.

Added "stand-alone" to all headings that describe stand-alone
installation. Just to make it extra clear.

Moved logging deps config to tools deps usage section as it only
applies to that usage.

Re-orged Clojure Tools instructions a bit.
Clarified, I think, how it processes args.

Noticed that JVM options only applies to stand-alone variants to added
that detail. I also noticed that it does not give an example, which I
thought would be useful, but did nothing about that.

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
+== Use as dependency
 
-An alternative is to give a `sha`, which allows us to use old versions or versions that hasn't been released yet, e.g.:
+An alternative way of executing the `poly` tool is to specify it as a dependency in your `./deps.edn` file.
```
Most sentences use `our` instead of `your`, or `we` instead of `you`, as if we did this together. Change this to make it consistent.

lread replied:
Yes, from my (admittedly huge) commit comment:

> Subtle: tried to use "we" only when talking from Polylith core team, and used "you" when talking to and guiding the reader.

My personal feeling is using "our" and "we" is a bit confusing and makes me sometimes wonder what "I" should be doing. It's subtle. I just did a web search and found this:
https://bernoff.com/blog/the-pronoun-conundrum-we-vs-you-in-business-advice

This is subjective and will adhere to your preference, lemme know.

tengstrand replied:
Hmm, interesting! This is important, so let me check this and come back to you!

tengstrand commented on doc/install.adoc:
```diff
@@ -3,59 +3,80 @@
 
 [#install-on-mac]
-== Install on Mac
+== Install as Stand-alone on macOS
```
Do we need to use upper case for Stand-alone? I prefer stand-alone, if you don't have any strong opinions.

lread replied:
No strong opinion at all, but capitalization for section titles should probably be consistent, whatever convention you choose.

lread replied:
Maybe it is nice to capitalize headings "Like normal sentences"? I think that is fine too, lemme know.

tengstrand replied:
Yes, capitalizing the headings, but lower case elsewhere.

lread replied:
So... to make sure I got it: for headings you'd like for example:

== The Dog and Cat are Friends

Instead of 

== The dog and cat are friends

(So Stand-alone would remain capitalized?)
I'm fine either way. Lemme know.

tengstrand commented on doc/install.adoc:
```diff
@@ -3,59 +3,80 @@
-2. By the https://clojure.org/guides/deps_and_cli[Clojure CLI], e.g. `clojure -M:poly help`:
-* The `poly` tool can only be executed from inside a Polylith xref:workspace.adoc[workspace] directory (it uses the `:poly` alias in `./deps.edn`).
-* Which version of the `poly` tool to use can easily be selected by changing the `sha` in `./deps.edn`.
+| Example usage
```
The text in this column (for all rows in the table) could be bold (to match the top headers) and end with a `:` e.g. **Example usage:**

lread replied:
Sure, will do!

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
-[source,shell]
+{post-install}
+
+=== Testing from Master
```
Change to "Run from master" or similar, because many users use the latest commit in master to get access to the latest functionality (not only for testing). Maybe Master can be in lower case, because the name of that branch is `master`?

lread replied:
Good point, I did not mean to use "Testing" will switch as recommended.
See my previous question for section title capitalization.

Aside: I expect your repo was long created before https://www.bbc.com/news/technology-53050955, but some readers might be sensitive to the term master. Not making a suggestion here, just a comment.

tengstrand replied:
Good point. The default is `main` when you create a new Polylith workspace. I think we kept it to reduce complexity, don't remember (we had a discussion in the Polylith team).

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
 
-We can get basic built-in help via the CLI's help machinery:
+An alternative to specifying a `:git/tag` is to specify a `:git/sha`.
+This allows you to install any git revision poly and is often used to test out a version that has not been officially released yet:
```
I try to use `us` (and we) when possible instead of `you`, as if we are doing this together.

lread replied:
See previous comment on us and we.

tengstrand commented on doc/install.adoc:
```diff
@@ -3,59 +3,80 @@
+:prerequisites: First, ensure that you have https://clojure.org/guides/getting_started[Clojure CLI] and https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[git] installed. +
+If you are installing git for the first time, don't forget to configure your git https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git[user.name] and https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git[user. ...
+
+:post-install: pass:n[You will soon create your first xref:workspace.adoc[workspace] - for now, only `version`, `help`, `doc`, and `create workspace` commands will work.]
```
Change to "We will soon create our first...". I try to use `we` instead of `you` when possible, to emphasise that we are doing this together! (check the other occurrencies too)

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
+The command-line argument syntax for Clojure CLI tools uses an "exec args" format, essentially Clojure keyword/value pairs.
 
-[source,clojure]
+For this reason, it can be convenient to first launch a `poly` shell so you can use the conventional `poly` command-line argument syntax.
```
`we` instead of `you`.

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
 
-[source,clojure]
+For this reason, it can be convenient to first launch a `poly` shell so you can use the conventional `poly` command-line argument syntax.
+Let's say you want to get info on lines of code.
```
`we` instead of `you`

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
 
-[#github-dependency]
-Via GitHub:
+After which you can specify standard Polylith arguments:
```
`we`

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
+After which you can specify standard Polylith arguments:
+[source,shell,subs="attributes+"]
+----
+                  _      _ + _   _
```
The ascii-art doesn't render nicely for me. Some characters are black, others are grey. One solution could be to change from `shell` to `text`. What is `attributes+` doing?

lread replied:
Good catch. Will try changing code style to text.

See the commit comment for `attributes+`. This allows adoc attributes to be replaced in code block, in this case {poly-version}.

tengstrand commented on doc/install.adoc:
```diff
@@ -148,138 +176,170 @@ nix-shell -p polylith
-                                org.apache.logging.log4j/log4j-slf4j-impl {:mvn/version "2.20.0"}}}}
-----
+You might want to specify more RAM to the `poly` tool and/or where the configuration file for logging is located.
+These can be conveyed via the `JVM_OPTS` environment variable which `poly` will passes along to the Java runtime at launch of poly stand-alone. 
```
I think the logging example is relevant, for not-techie users (as me!).

lread replied:
See commit comment:

> Moved logging deps config to tools deps usage section as it only applies to that usage.

tengstrand replied:
Okay, cool.

lread committed:
review feedback: header style for 1st table col

And append colon to 1st col text.

lread committed:
review feedback: Running from master

Also got around normal heading capitalization convention by quoting
master as code.

lread committed:
review feedback: text as lang for poly shell block

lread committed:
Restore link to poly release strategy

I accidentally deleted this.

tengstrand reviewed (approved):
This looks really good and all changes make sense to me! I haven't check with my "English teacher friend" yet regarding the use of you/we, but I just decided to go with "your" style for now and use `you` as default. It's really hard to decide when to use `we` or `you`, so having `you` as default kind of makes sense because it's "never wrong" which `we` sometimes is. You can use `we` whenever that feels convenient, but let's figure this out later!
